### PR TITLE
Change pub fields into functions.

### PR DESCRIPTION
**Pull request by ahmedcharles**
This hides the implementation enough to allow supporting higher memory kernels.

### CODE REVIEW

File: src/elf_sections.rs [hunk 1]
```diff
@@ -82,17 +82,34 @@ impl Iterator for ElfSectionIter {
...
+            3 => ElfSectionType::StringTable,
+            4 => ElfSectionType::RelaRelocation,
+            5 => ElfSectionType::SymbolHashTable,
+            6 => ElfSectionType::DynamicLinkingTable,
+            7 => ElfSectionType::Note,
+            8 => ElfSectionType::Uninitialized,
+            9 => ElfSectionType::RelRelocation,
+            10 => ElfSectionType::Reserved,
+            11 => ElfSectionType::DynamicLoaderSymbolTable,
+            _ => panic!(),
```
Comment (4e554c4c) [inline src/elf_sections.rs]: Perhaps it would not be best to panic here because environment and processor specific flags are allowed.

**Comment by ahmedcharles** [comment]
Updated.

**Comment by 4e554c4c** [comment]
Hey, since I don't have write access can you update this commit with the most recent master?

**Comment by ahmedcharles** [comment]
I was just doing that and I made a comment on the other commit, since I think it can be implemented at runtime, which would be significantly nicer.

**Comment by ahmedcharles** [comment]
It's updated.

**Comment by ahmedcharles** [comment]
That would prevent ElfSectionType from being `#[repr(u32)]`. I suspect it would be better to just write a custom type, but I'd rather do that in a separate PR. I've added `section_type_raw()` as a workaround, so that information isn't lost in the mean time.

**Comment by 4e554c4c** [comment]
Looks good to me 👍

**Comment by le-jzr** [comment]
wrt ElfSectionType, repr(u32) is largely pointless here. We cannot convert from u32 to ElfSectionType, because that's undefined behavior, and the opposite direction we don't need.

**Pull request merged**